### PR TITLE
VR comes to the web, part II.

### LINKED ISSUE #1810: content: WebXR, part II: the frame loop
**Author**: @jpmedley
**Reviewer**: [Leave blank. web.dev team will assign a reviewer.]

**Target publish date:** <yyyy-mm-dd>
- [ ] Check this box if this is a hard deadline.

- [ ] Check this box if you'd like your reviewer to fix grammar and punctuation issues for you.

**Process**

Check out the [web.dev writing process](https://web.dev/handbook/quick-start/).

When you're ready, make a copy of [this template](https://docs.google.com/document/d/1lgaNIEnXZf-RB8_p9RK22QEgpXJqnu77pLWVWVy4nuw/edit)
and fill in the proposal and outline sections. When you're finished, drop a link
to the doc in the 1st step below and check the box. A member of the web.dev team
will assign this ticket to themselves and work as your reviewer to help you
complete the remaining steps.

- [x] 1. I've done my proposal <add a link to your proposal doc>
- [ ] 2. Proposal approved — _reviewer checks this_
- [x] 3. I've added [a draft to my proposal doc](https://docs.google.com/document/d/1lgaNIEnXZf-RB8_p9RK22QEgpXJqnu77pLWVWVy4nuw/edit)
- [ ] 4. Draft approved — _reviewer checks this_
- [ ] 5. I've submitted a pull request #<add PR number>
- [ ] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by jpmedley**
Fixes #1810 

This PR also contains updates to part I to make it work better with part II. Specifically, it moves all content related to frame loops into the second article.

### CODE REVIEW

File: src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+---
+title: Virtual reality comes to the web, part II
+subhead: All about the frame loop
+authors:
+  - joemedley
+date: 2020-02-12
+hero: hero.jpg
+alt: A person using a virtual reality headset.
+description:
+  Virtual reality came to the web in Chrome 79. Based on the WebXR Devicer API,
```
Comment (kaycebasques) [inline src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md]:
```suggestion
  Virtual reality came to the web in Chrome 79. Based on the WebXR Device API,
```

File: src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+also provided instructions for requesting, entering, and ending an XR session.
+
+This article describes the frame loop, which is a user-agent controlled
+infinite loop in which content is repeatedly drawn to the screen. Content is
+drawn in discrete blocks called frames. The succession of frames creates the
+illusion of movement.
+
+## What this article is not
+
+WebGL and WebGL2 are the only means of rendering content during a frame loopß in
```
Comment (kaycebasques) [inline src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md]:
```suggestion
WebGL and WebGL2 are the only means of rendering content during a frame loop in
```

File: src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+of WebGL and WebGL2. Such frameworks include [three.js](https://threejs.org/),
+[babylonjs](https://www.babylonjs.com/), and
+[PlayCanvas](https://playcanvas.com/), while [A-Frame](https://aframe.io/) and
+[React 360](https://facebook.github.io/react-360/) are designed for interacting
+with WebXR.
+
+This article is neither a WebGL nor a framework tutorial. It explains basics of
+a frame loop using the Immersive Web Working Group's Immersive VR Session sample
+([demo](https://immersive-web.github.io/webxr-samples/immersive-vr-session.html)
+[source](https://github.com/immersive-web/webxr-samples/blob/master/immersive-vr-session.html).
```
Comment (kaycebasques) [inline src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md]: Formatting is off here

image

File: src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+  that later) you will never access this object directly. You will merely pass
+  it from the <code>XRWebGLLayer</code> to the <code>WebGLRenderingContext</code>.</dd>
+  <dt>XRViewport</dt>
+  <dd>A viewport provides the coordinates and dimensions of a rectangular region in
+  the <code>WebGLFramebuffer</code>.</dd>
+  <dt>WebGLRenderingContext</dt>
+  <dd>A rendering context is a programmatic access point for a canvas (the space
+  we're drawing on). To do this, it needs both a <code>WebGLFramebuffer</code> and an
+  <code>XRViewport</code>.</dd>
+</dl>
```
Comment (kaycebasques) [inline src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md]: `dl` doesn't seem well-supported on web.dev:

image

The path of least resistance is to just use different semantic elements

File: src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+  <dt><code>XRViewerPose</code<></dt>
+  <dd>A pose is the position and orientation of something in 3D space. Both viewers
+  and input devices have a pose, but it's the viewer's pose we're concerned with
+  here. Both viewer and input device poses have a <code>transform</code> attribute
+  describing its position as a vector and its orientation as a quaternion
+  relative to the origin. The origin is specified based on the requested
+  reference space type when calling <code>XRSession.requestReferenceSpace()</code>.</br></br>
+
+  Reference spaces take a bit to explain. I cover them in depth in [yet to be
+  named AR article](link here). The sample I'm using as the basis for this article
```
Comment (kaycebasques) [inline src/site/content/en/blog/vr-comes-to-the-web-pt-ii/index.md]: TODO missing link

**Comment by jpmedley** [comment]
@kaycebasques Feedback addressed. PTAL

**Comment by jpmedley** [comment]
@kaycebasques  The ball is back in your court.

**Comment by jpmedley** [comment]
@kaycebasques PTAL

**Comment by kaycebasques** [comment]
@jpmedley it seems like you want to coordinate the release of this with the Hit Test article, so I'm not going to merge it. Because the PR is approved you should be able to merge it when you see fit.

**Pull request merged**